libs/order/src/drivers/src/magento/order.service.spec.ts

```diff
@@ -0,0 +1,455 @@
+
+          beforeEach(() => {
+            notOrderId = 'notOrderId';
+          });
+
+          it('should throw a DaffOrderNotFoundError', done => {
+            const expected = new DaffOrderNotFoundError(`Could not find an order with ID ${notOrderId}`);
+
+            service.get(notOrderId, cartId).subscribe(result => {
+              fail();
+              done();
+            }, error => {
+              expect(error).toEqual(expected);
+              done();
+            });
+
+            const op = controller.expectOne(getGuestOrders);
+
+            op.flush({
+              data: mockGetOrdersResponse
+            });
+          });
+        });
+      });
+    });
+
+    describe('when the call to the Magento API is unsuccessful', () => {
+      it('should throw a DaffCartNotFoundError', done => {
+        service.get(orderId, cartId).pipe(
+          catchError(err => {
+            expect(err).toEqual(jasmine.any(DaffCartNotFoundError));
+            done();
+            return [];
+          })
+        ).subscribe();
+
+        const op = controller.expectOne(getGuestOrders);
+
+        op.graphqlErrors([new GraphQLError(
+          'Can\'t find a cart with that ID.',
+          null,
+          null,
+          null,
+          null,
+          null,
+          {category: 'graphql-no-such-entity'}
+        )]);
+      });
+    });
+  });
+
+  describe('list | listing the available orders', () => {
+    describe('when the call to the Magento API is successful', () => {
+      describe('and the response passes validation', () => {
+        beforeEach(() => {
+          validatorSpy.and.returnValue({data: mockGetOrdersResponse});
+        });
+
+        it('should return the list of Daffodil orders', done => {
+          service.list(cartId).subscribe(result => {
+            expect(result).toEqual([jasmine.objectContaining(mockDaffOrder)]);
+            done();
+          });
+
+          const op = controller.expectOne(getGuestOrders);
+
+          op.flush({
+            data: mockGetOrdersResponse
+          });
+        });
+      });
+
+      describe('and the response fails validation', () => {
+        beforeEach(() => {
+          validatorSpy.and.callFake(() => {
+            throw new DaffInvalidAPIResponseError('Get orders response does not contain a valid list of orders.')
+          });
+        });
+
+        it('should throw a DaffInvalidAPIResponseError', done => {
+          service.list(cartId).pipe(
+            catchError(err => {
+              expect(err).toEqual(jasmine.any(DaffInvalidAPIResponseError));
+              done();
+              return [];
+            })
+          ).subscribe();
+
+          const op = controller.expectOne(getGuestOrders);
+
+          op.flush({
+            data: mockGetOrdersResponse
+          });
+        });
+      });
+    });
+
+    describe('when the call to the Magento API is unsuccessful', () => {
+      it('should throw a DaffCartNotFoundError', done => {
```

Comment:
I don't think you need to test the error transform logic here, but I think this "it" block should just be "should throw an error" or something more general. Just to be clear that this won't necessarily throw a DaffCartNotFoundError.

Reply:
changed